core/controllers/editor.py

```diff
@@ -190,10 +190,12 @@ def get(self, exploration_id: str) -> None:
         # 'apply_draft' and 'v'(version) are optional parameters because the
         # exploration history tab also uses this handler, and these parameters
         # are not used by that tab.
-        assert self.user_id is not None
-        assert self.normalized_request is not None
-        version = self.normalized_request.get('v')
-        apply_draft = self.normalized_request['apply_draft']
+        if self.normalized_request:
+            version = self.normalized_request.get('v')
+            apply_draft = self.normalized_request.get('apply_draft')
```

Comment:
Here we are sure that 'v' and 'apply_draft' always going to be present, right ?

If yes, Then please remove .get and access values directly using `[]`.

Reply:
Also please take a look at the failing tests.

Reply:
‘v’ might not always be present in self.normalized_request. It’s safer to use .get() method to avoid KeyError.